bookkeeper-server/src/test/java/org/apache/bookkeeper/client/MockBookKeeper.java

```diff
@@ -210,6 +213,66 @@ public void close() throws InterruptedException, BKException {
         shutdown();
     }
 
+    @Override
+    public OpenBuilder newOpenLedgerOp() {
+        return new OpenBuilder() {
+            long ledgerId;
+            byte[] password;
+            org.apache.bookkeeper.client.api.DigestType digestType;
+
+            @Override
+            public OpenBuilder withLedgerId(long ledgerId) {
+                this.ledgerId = ledgerId;
+                return this;
+            }
+
+            @Override
+            public OpenBuilder withRecovery(boolean recovery) {
+                return this;
```

Comment:
I guess this will be enhanced when we will implement a new mock WriteHandle

Reply:
if needed. currently the mock calls asyncOpenLedger for asyncOpenNoRecovery. 

Originally this code comes from pulsar, where they don't use tailing anyhow.